internal/backend/azure/azure.go

```diff
@@ -207,25 +207,41 @@ func (be *Backend) Path() string {
 	return be.prefix
 }
 
+// useAccessTier determines whether to apply the configured access tier to a given file.
+// It returns true if the configured AccessTier is not "archive" and the file is a non-metadata pack file.
+// This ensures that:
+// 1. No files are ever stored in the Archive tier, regardless of the AccessTier setting.
+// 2. Only data files (non-metadata pack files) use the configured AccessTier when it's not "archive".
+func (be *Backend) useAccessTier(h backend.Handle) bool {
+	noArchiveClass := strings.ToLower(be.cfg.AccessTier) != "archive"
+	isDataFile := h.Type == backend.PackFile && !h.IsMetadata
+	return isDataFile && noArchiveClass
+}
+
 // Save stores data in the backend at the handle.
 func (be *Backend) Save(ctx context.Context, h backend.Handle, rd backend.RewindReader) error {
 	objName := be.Filename(h)
 
 	debug.Log("InsertObject(%v, %v)", be.cfg.AccountName, objName)
 
+	var accessTier blob.AccessTier
+	if be.useAccessTier(h) {
+		accessTier = blob.AccessTier(be.cfg.AccessTier)
+	}
+
 	var err error
 	if rd.Length() < saveLargeSize {
 		// if it's smaller than 256miB, then just create the file directly from the reader
-		err = be.saveSmall(ctx, objName, rd)
+		err = be.saveSmall(ctx, objName, rd, &accessTier)
 	} else {
 		// otherwise use the more complicated method
-		err = be.saveLarge(ctx, objName, rd)
+		err = be.saveLarge(ctx, objName, rd, &accessTier)
 	}
 
 	return err
 }
 
-func (be *Backend) saveSmall(ctx context.Context, objName string, rd backend.RewindReader) error {
+func (be *Backend) saveSmall(ctx context.Context, objName string, rd backend.RewindReader, accessTier *blob.AccessTier) error {
```

Comment:
Currently that parameter is never nil. Why pass it as pointer then?

Reply:
Done, fixed.